src/hotspot/cpu/x86/x86.ad

```diff
@@ -1773,17 +1773,9 @@ const bool Matcher::match_rule_supported_vector(int opcode, int vlen, BasicType
       }
       break;
     case Op_VectorCastB2X:
-      if (size_in_bits == 256 && UseAVX < 2) {
-        return false; // Implementation limitation
-      }
-      break;
     case Op_VectorCastS2X:
-      if (is_integral_type(bt) && size_in_bits == 256 && UseAVX < 2) {
-        return false;
-      }
-      break;
     case Op_VectorCastI2X:
-      if (is_integral_type(bt) && size_in_bits == 256 && UseAVX < 2) {
+      if (bt != T_DOUBLE && size_in_bits == 256 && UseAVX < 2) {
```

Comment:
CastI2X  should work with (UseAVX == 1) for (bt == T_FLOAT) so the prior code was correct for CastI2X. 
The fix is only needed for  CastS2X and CastB2X.

Reply:
AFAIK int vectors of size 256 bit are not supported on AVX1 so we cannot cast from int vectors to float vectors of size 256 anyway. So it is essentially the same.

Reply:
x86 docs say that `vcvtdq2ps` is AVX1 instruction:
![image](https://user-images.githubusercontent.com/5215794/147149423-10988f8d-9bcd-41ca-b728-bc2c34e3bd35.png)

Reply:
Hi, while the instruction is supported in hardware, the VM does not support the vector shape. As a result, there should not be a situation where we have a vector node with that shape on AVX1. Thank you very much.

Reply:
Why do you think that? What about this code when length is the same (and UseAVX could be 1):
https://github.com/openjdk/jdk18/blob/master/src/hotspot/cpu/x86/x86.ad#L7069

Reply:
Because the input of `VectorCastI2XNode` would be a vector node of type int and size 256, this shape is not supported on AVX1. So, we should not have this node because its input should not appear.

Reply:
Right.